doc/source/how-to-use-clientapp.rst

```diff
@@ -0,0 +1,303 @@
+Use :code:`ClientApp`
+=====================
+
+With Flower 1.8, we introduced the :code:`ClientApp` and :code:`ServerApp`. These are new features in Flower that allow more flexibility for you to run your federated learning projects, such as adding intermediate operations in your workflows :doc:`using Mods <how-to-use-built-in-mods>`, maximizing compute usage by :doc:`running multiple projects concurrently <how-to-use-multirun>`, and speeding up your project deployment process by :doc:`deploying from a simulation setup <how-to-deploy-to-production>`.
+
+.. |clientapp_link| replace:: ``ClientApp()``
+.. |serverapp_link| replace:: ``ServerApp()``
+.. |message_link| replace:: ``Message()``
+.. |context_link| replace:: ``Context()``
+.. |flower_clientapp_link| replace:: ``flower-client-app``
+.. _clientapp_link: ref-api/flwr.client.ClientApp.html
+.. _serverapp_link: ref-api/flwr.server.ServerApp.html
+.. _message_link: ref-api/flwr.common.Message.html
+.. _context_link: ref-api/flwr.common.Context.html
+.. _flower_clientapp_link: ref-api-cli.html#flower-client-app
+
+What is a :code:`ClientApp`?
+----------------------------
+
+The |clientapp_link|_ is an application that can run various tasks on clients, such as model training. It is launched on demand by the :code:`SuperNode`. The communication pattern between the :code:`SuperNode` and :code:`ClientApp` uses :code:`Message` and :code:`Context` objects containing all the information needed to run the tasks. These objects will be elaborated further below.
+
+There are two ways to use the :code:`ClientApp`:
+
+1. :ref:`Using high-level ClientApp API<Using high-level API>`
+2. :ref:`Using low-level ClientApp API<Using low-level API>`
+
+Using high-level API
+--------------------
+
+This approach is the compatibility mode. It allows us to reuse any client-code that has been implemented using :code:`NumpyClient` or :code:`Client` classes with a few additional lines of code. Let’s take the :doc:`PyTorch Quickstart <tutorial-quickstart-pytorch>` code for example. In that example, we have included a :code:`CifarClient` class that inherits from the :code:`NumpyClient` convenience class:
```

Comment:
I'm wondering if it would be worth linking to the "migration guide" page you added recently?  maybe at the bottom of this section. e.g. "please refer to <migration guide> to see how to migrate nos just ClientApp but also ServerApp and your Simulation". Would something like this make sense? (maybe as a admotion note or similar)

Reply:
Good idea, I've added an admonition at the start of this section.